### PR TITLE
Migrate reviews panel to home screen

### LINKED ISSUE #5240: Home Screen - migrate Reviews panel
**User story:**
As a merchant
I want to easily see how many new reviews my store has when I enter WooCommerce
So I can review them as soon as possible

**Description:** we’re going to migrate the reviews panel features to the home screen, as managing reviews can be an important task for stores with Product Reviews enabled

**Acceptance criteria:**
- [ ] Display a Product Reviews task in the home screen in stores with [Product Reviews](https://docs.woocommerce.com/document/configuring-woocommerce-settings/#section-9) enabled
- [ ] Display up to 5 reviews pending moderation

- [ ] The review card should include the following elements: image, product name, customer name, rating, comment, time and the `approve`, `mark as spam` and `delete` actions
- [ ] The user is redirected to the product page when clicking in the product name and redirected to the Customer report filtered by the customer name if the user clicks in the customer name
- [ ] Include a `Manage all reviews` CTA at the bottom of the task that redirects the user to the reviews list
- [ ] The review activity panel should be removed from the navbar once we have the same feature in the home screen

**Event tracking:**
Can we reuse the same event tracking that we have in the reviews panel?
We also need to add on to cover the moderate actions:
`wcadmin_reviews_action`
Event prop: `action`- `approve` |`mark as spam` | `delete`

**Additional links:**
[P2 with links to the Figma file](pbIJXs-ma-p2)

### PR DESCRIPTION
**Pull request by louwie17**
Fixes #5240 

The bulk of the code is migrating the reviews panel to the home screen. Currently there is a decent amount of overlapping logic between the reviews header panel and the home screen panel.
The home screen panel contains several actions (approve, set as spam, delete), these actions had to be added to the reviews data store - I added the updateReview and deleteReview actions.
Made use of `invalidateResolutionForStoreSelector` to re-trigger updates after update or deletion.

Updated the rating component to allow for custom icons to be passed in (as strings), this way I could show both the normal star, and the outline star as shown in the wireframe.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)

### Screenshots

review-panel

### Detailed test instructions:

- Create new store and add several products, make sure you have the [products reviews](https://docs.woocommerce.com/document/configuring-woocommerce-settings/#section-9) enabled.
- Go to the public product page and add some comments (in incognito so the comments show up as 'pending')
- Go to the Woo home screen
- You should see a `Reviews to moderate` on the left, with the number of pending reviews (Reviews header panel should not be shown - awaiting feedback if just on home screen)
- You should be able to Approve, mark as spam, or delete a review, a popup will appear that allows you to 'undo' this action
- If no reviews are needing review it should not be allowed to collapse.

UPDATE:
There are a couple small styling changes that depend on Jeff's updates: https://github.com/woocommerce/woocommerce-admin/pull/5729/files#diff-beff577e21ba5ae2d55e05bdfcbcc94c7da812eb81f26b5da415b870d79d2c82
I believe the only thing that is noticeable is the margin between the icon and the subtitle.

### CODE REVIEW

File: client/header/activity-panel/activity-card/style.scss
Comment (louwie17) [inline client/header/activity-panel/activity-card/style.scss:265]: Moved some of this to the homescreen reviews style.scss and deleted the rest as the header review panel is removed.

File: client/header/activity-panel/index.js
Comment (louwie17) [inline client/header/activity-panel/index.js:23]: Removing any use of the ReviewsPanel

File: client/homescreen/activity-panel/reviews/index.js
Comment (louwie17) [inline client/homescreen/activity-panel/reviews/index.js:160]: Showing loading placeholder when an review is updating or when it is updating the entire list.

File: client/homescreen/activity-panel/reviews/index.js
Comment (louwie17) [inline client/homescreen/activity-panel/reviews/index.js:396]: Clearing the `getReviewsTotalCount` if no reviews left, only necessary in the case we revert the last review. Without this it wouldn't update the home screen.

File: client/homescreen/activity-panel/style.scss
Comment (louwie17) [inline client/homescreen/activity-panel/style.scss:13]: This is overlapping css between the components button and the header.
css for the empty card-header.
Although maybe this css should be moved to the accordion component?

File: client/homescreen/activity-panel/index.js
Comment (louwie17) [inline client/homescreen/activity-panel/index.js:52]: Set collapsible to `false` if the count is 0

File: packages/components/src/rating/index.js
Comment (louwie17) [inline packages/components/src/rating/index.js:84]: This allows me to use the filled star and the outline star:
Screen Shot 2020-11-23 at 4 24 18 PM

File: client/homescreen/activity-panel/panels.js
Comment (samueljseay) [inline client/homescreen/activity-panel/panels.js:43]: It wouldn't have been obvious because I think it was buried in another issue discussion but the copy for this should now just be "Reviews"

File: client/homescreen/activity-panel/style.scss
Comment (samueljseay) [reply client/homescreen/activity-panel/style.scss:13]: It could make sense to,but I'll leave it up to you, if you wanna do that as a follow up later.

File: packages/components/src/accordion/test/index.js
Comment (samueljseay) [inline packages/components/src/accordion/test/index.js:53]: Looking over the tests you've added they look great. Good stuff!

**Comment by samueljseay** [comment]
Reviewing the look and feel it looks like some of the spacings and sizings are off from the new designs.

For example:

* In the new design the product image is 33x33px
* In the new design the right side margin of the product image is 16px, looks to be about 20 in yours
* Even with a short product name/description the stars on yours will run on to the next line. I wonder if we should truncate the text and make sure the stars always display on the same line @elizaan36 any input on how we'd handle a long line of product / reviewer name?
* In the new design the time since review was posted is no longer all caps.

There are a few other things, I'm also considering maybe it just wasn't clear that there was some redesign work to do here, so sorry about any ambiguity there. You can check out the designs (figma links at bottom) [here](pbIJXs-ma-p2-p2).

**Review by samueljseay — DISMISSED** [review]
Overall everything looks great, I think the reviews themselves need some redesign applied and there is a piece of copy to change, but otherwise absolutely stellar work. ✨

File: packages/components/src/accordion/test/index.js
Comment (louwie17) [reply packages/components/src/accordion/test/index.js:53]: > There are a few other things, I'm also considering maybe it just wasn't clear that there was some redesign work to do here, so sorry about any ambiguity there. You can check out the designs (figma links at bottom) [here](pbIJXs-ma-p2-p2).

Thanks I did look at this before and I probably should off asked how to get the specific pixel amounts and such. I just noticed I never saw them as I didn't create an account for Figma.

File: client/homescreen/activity-panel/style.scss
Comment (louwie17) [reply client/homescreen/activity-panel/style.scss:12]: Yeah I suppose the same goes for the activity card, we will probably have to consolidate the css once @jeffstieler stock migration is finished as well, given the Activity Card will largely be used on the home screen instead of the header. It looks like the Inbox would be the only one still using it.

**Comment by louwie17** [comment]
Updated styling changes mentioned by @samueljseay in https://github.com/woocommerce/woocommerce-admin/pull/5706/commits/bbd5e69f38b9132daa080986da8c175b9140483b
These changes did leave some more room between the title, rating and the date, awaiting feedback incase of truncation of the title.

File: client/homescreen/activity-panel/reviews/index.js
Comment (jeffstieler) [reply client/homescreen/activity-panel/reviews/index.js:400]: This should be unnecessary after https://github.com/woocommerce/woocommerce-admin/pull/5729/commits/9d7626d558a5b0b2a1889d73bea039b4a78df6fe lands.

File: packages/data/src/reviews/actions.js
Comment (jeffstieler) [inline packages/data/src/reviews/actions.js:22]: What do you think about creating some status-specific actions like `markReviewAsSpam()` and `approveReview()` so that the component code can be a little simpler?

File: client/homescreen/activity-panel/reviews/index.js
Comment (jeffstieler) [inline client/homescreen/activity-panel/reviews/index.js:77]: Why is this `2`? Doesn't the panel show `5` reviews?

File: client/homescreen/activity-panel/reviews/index.js
Comment (louwie17) [reply client/homescreen/activity-panel/reviews/index.js:400]: @jeffstieler nice, would this require me to use `getItemsTotalCount` instead of `getReviewsTotalCount`?

File: client/homescreen/activity-panel/reviews/index.js
Comment (louwie17) [reply client/homescreen/activity-panel/reviews/index.js:77]: Yeah it shows 5, this was in reference to this change: https://github.com/woocommerce/woocommerce-admin/pull/5706#discussion_r528972306 to only call it if it was the last item being updated.
I could probably check if `reviews.length` would actually be zero (I didn't specifically test that, as I thought it might show as `1` still within the scope).

File: client/homescreen/activity-panel/style.scss
Comment (elizaan36) [reply client/homescreen/activity-panel/style.scss:12]: in case this factors in, just adding a note here that we're going to remove the Inbox from the app header altogether so it will only appear on the Home screen.

**Comment by elizaan36** [comment]
Just chatted with @louwie17 on Slack a bit and we're going to move the star rating to the next line below the customer and product name. In the case of a long product or customer name, the line will wrap. @jeffstieler already has this implemented in his PR so maybe this is another thing to consolidate.

Here's the design with star rating on a new line.
Screen Shot 2020-11-24 at 12 39 14 PM

I also noticed some minor style tweaks that we talked about on Slack, mainly to do with the card header. But I'll wait until one or the other panels is merged to have a look at those styles again.

File: client/homescreen/activity-panel/reviews/index.js
Comment (jeffstieler) [reply client/homescreen/activity-panel/reviews/index.js:400]: I don't think so as their implementations are identical and the total update happens in the resolver.

File: packages/data/src/reviews/actions.js
Comment (louwie17) [reply packages/data/src/reviews/actions.js:22]: Yeah not a bad idea, the only caveat was 'reverting' a change, which made having a generic `update` function rather handy. I could do both. In either case would it make sense to move the `createNotice` to the action as well?

Comment (jeffstieler) [reply packages/data/src/reviews/actions.js:22]: Ah right.. the previous status won't always be the same making reversal unable to take advantage of the "simpler" actions. Let's scrap the idea 🙃 .

> In either case would it make sense to move the `createNotice` to the action as well?

No I don't think we'd want to couple those in the event that UX needed to change or the action could be used in multiple different contexts.

File: packages/components/src/accordion/test/index.js
Comment (samueljseay) [reply packages/components/src/accordion/test/index.js:53]: @louwie17 did you get sorted out with Figma access? There are a few tools in there for measuring distance between objects in the designs, let me know if you need help with any of that.

Comment (louwie17) [reply packages/components/src/accordion/test/index.js:53]: @samueljseay I did figure that out, although a little overview of the tools and how to's would certainly be helpful. We can do that during our 1:1 to tomorrow?

**Comment by louwie17** [comment]
Ready for review again, fixed merge conflicts with https://github.com/woocommerce/woocommerce-admin/pull/5729 PR, @elizaan36 I believe all the styling changes reflect the wireframe now and the rating is moved down.

@jeffstieler I noticed that non-refresh issue again, after undoing the last item. Turns out it has to do with this change I did - only showing the header: https://github.com/woocommerce/woocommerce-admin/pull/5706/commits/ebd963fc48d07dfa5afccaea2b43ef70a3f7ff02 as it won't render the actual panel in that case and not re-trigger the selector.
I was going to use the same fix I did on yours, but noticed that the `invalidateResolution` calls actually don't do anything (you can see that there are no changes in the screenshot):
Screen Shot 2020-11-25 at 5 26 08 PM

I didn't want to call `invalidateResolutionForStoreSelector( 'getItems' );` as that would clear out `orders` as well. Any thoughts? Should we create specific selectors for the types -> `getOrders`, `getProducts`, etc...?
This can probably be done in a separate PR.

**Comment by jeffstieler** [comment]
@louwie17 it looks like the `collapsible` change is causing some weird behavior with the stock panel as well:

Capture 2020-11-25 at 16 49 01

Here's what it looks like for both the stock and review panels without setting the `collapsible` prop:

Capture 2020-11-25 at 16 51 38

**Comment by louwie17** [comment]
> @louwie17 it looks like the `collapsible` change is causing some weird behavior with the stock panel as well:

@jeffstieler do you mean the drop down arrow disappearing? Once there are zero items it just renders a header as displayed in the first picture here: https://github.com/woocommerce/woocommerce-admin/issues/5239#issuecomment-729897323
Which is what the `collapsible` thing is for, although I can certainly scratch that (or just hide the collapse with css).

File: client/homescreen/activity-panel/reviews/index.js
Comment (jeffstieler) [inline client/homescreen/activity-panel/reviews/index.js:81]: Do these have to be closures? Can we just pass `clearReviewsCache`?

File: client/homescreen/activity-panel/stock/index.js
Comment (jeffstieler) [inline client/homescreen/activity-panel/stock/index.js:61]: Why is the Stock panel being modified in this PR? If there's an issue with the Stock panel, can you open a separate PR?

**Review by jeffstieler — APPROVED** [review]
This looks good, just one minor callback nitpick. However, I do think the Stock panel changes should be removed from this PR and moved to a new one.

File: client/homescreen/activity-panel/stock/index.js
Comment (louwie17) [reply client/homescreen/activity-panel/stock/index.js]: Reverted the last commit and took out [the above changes](https://github.com/woocommerce/woocommerce-admin/pull/5706/commits/1a01312b54acbc0db9eab2fb4abb87fcbe469d96), these are no part of a different branch/PR -> https://github.com/woocommerce/woocommerce-admin/commit/16fb47b8b601c938ee58d32eb975f5b0a9265c45

**Comment by jeffstieler** [comment]
I spot checked this after the removal of the Stock panel changes - looks good! :shipit:

**Pull request merged**
